osutil/mountinfo.go

```diff
@@ -0,0 +1,30 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"io"
+	"os"
+)
+
+var openMountInfoFile = func() (io.ReadCloser, error) {
+	mustNotBeTestBinary("/proc/self/mountinfo must be mocked in tests")
+	return os.Open("/proc/self/mountinfo")
```

Comment:
nitpick: there is something a bit confusing in having this linux specific code in the non _linux file. It can probably be cleaned up in a follow up.

Reply:
I can propose defining a constant value in a Linux mountinfo file and other OS-oriented mountinfo files, a constant value that specifies the path to a file that is relevant to the compiling OS. And the function here will simply use this constant value to open the file, whilst staying decoupled from the system.
@pedronis, please advise?

Reply:
it seems to me that this function is defined here only to satisfy the macOS building requirements, however we don't run tests on macOS, we just want to be able to build the very limited in functionality `snap` command on macOS, so I think it's more "in line" with the rest of the codebase to instead move the linux specific implementation of this file as you have it over to mountinfo_linux.go and then delete this mountinfo.go file and instead create a new mountinfo_darwin.go file (named just like that) defined like this:
```go

// license stuff

package osutil

import "io"

// openMountInfoFile is not implemented on darwin
var openMountInfoFile = func() (io.ReadCloser, error) {
    return nil, ErrDarwin
}
```

some day maybe we could join all of the nfs_darwin.go, mountinfo_darwin.go, etc all into one osutil_darwin.go but that doesn't need to be done in this PR

Reply:
I see your point. Indeed this way it looks more consistent. I will refactor it according to your advice. Thank you.